### DISCUSSION TITLE
Install API header file via install(target)

### DISCUSSION THREAD

**Discussion by ThomsonTan** [open-telemetry/opentelemetry-cpp#737] [General]
In today's SIG meeting, @eyakimov-bbg had a question that why the API header files are installed via `install(DIRECTORY)` instead of using the existing `install(TARGET)` as below. There is CMake target property `PUBLIC_HEADER` which is honored by `install(TARGET)`, but that does flat the directory structure of the header file. It seems `install(DIRECTORY)` is necessary to preserve the directory structure of our API header files.

https://github.com/open-telemetry/opentelemetry-cpp/blob/d8467a78b3cf3accfa3a0a2569bfd3b90822ad02/api/CMakeLists.txt#L16

Reference:
[StackOverflow](https://stackoverflow.com/questions/48212771/cmake-install-header-files-and-maintain-directory-heirarchy)

@owent @maxgolov @eyakimov-bbg @lalitb

**Comment by eyjohn** [comment]
Thanks for raising this one Thomas, This answers my initial question,
